types/address.go

```diff
@@ -550,103 +413,68 @@ func (ca *ConsAddress) Unmarshal(data []byte) error {
 	return nil
 }
 
-// MarshalJSON marshals to JSON using Bech32.
+// MarshalJSON marshals to JSON using hex.
 func (ca ConsAddress) MarshalJSON() ([]byte, error) {
 	return json.Marshal(ca.String())
 }
 
-// MarshalYAML marshals to YAML using Bech32.
+// MarshalYAML marshals to YAML using hex.
 func (ca ConsAddress) MarshalYAML() (interface{}, error) {
 	return ca.String(), nil
 }
 
-// UnmarshalJSON unmarshals from JSON assuming Bech32 encoding.
+// UnmarshalJSON unmarshals from JSON assuming hex encoding.
 func (ca *ConsAddress) UnmarshalJSON(data []byte) error {
 	var s string
-
-	err := json.Unmarshal(data, &s)
-	if err != nil {
+	if err := json.Unmarshal(data, &s); err != nil {
 		return err
 	}
-	if s == "" {
-		*ca = ConsAddress{}
-		return nil
-	}
 
-	ca2, err := ConsAddressFromBech32(s)
-	if err != nil {
-		return err
-	}
+	*ca = common.HexToAddress(s).Bytes()
 
-	*ca = ca2
 	return nil
 }
 
-// UnmarshalYAML unmarshals from YAML assuming Bech32 encoding.
+// UnmarshalYAML unmarshals from YAML assuming hex encoding.
 func (ca *ConsAddress) UnmarshalYAML(data []byte) error {
 	var s string
-
-	err := yaml.Unmarshal(data, &s)
-	if err != nil {
+	if err := yaml.Unmarshal(data, &s); err != nil {
 		return err
 	}
-	if s == "" {
-		*ca = ConsAddress{}
-		return nil
-	}
 
-	ca2, err := ConsAddressFromBech32(s)
-	if err != nil {
-		return err
-	}
+	*ca = common.HexToAddress(s).Bytes()
 
-	*ca = ca2
 	return nil
 }
 
 // Bytes returns the raw address bytes.
 func (ca ConsAddress) Bytes() []byte {
-	return ca
+	return ca[:]
```

Comment:
Not sure I get this change

Reply:
This should actually not be there, thanks for spotting it.
I guess I just forgot to remove while doing some local tests.
Will do.

Reply:
Reverted in latest commit

Reply:
Resolved